### PR TITLE
Query address-metadata from PFS endpoint before direct transfer

### PR DESCRIPTION
**Pull request by ezdac**
## Description
For a direct transfer, the node is not querying the PFS for a path. Since the main source of information for 
address-metadata is the Path-resource from the PFS, a direct transfer needs another means to retrieve the 
information. In this PR, the address-metadata endpoint of the PFS is accessed in order to 
get the address-metadata of the receiving node of a direct transfer.
Additionally, the sending nodes' address-metadata is injected in order for the receiver to 
interact with the sender easier.

Fixes: #6841

### CODE REVIEW

File: requirements/requirements-dev.in [hunk 1]
```diff
@@ -46,5 +46,5 @@ coverage
 bump2version
 
 # Test support
-matrix-synapse==1.27.0
+matrix-synapse==1.29.0
```
Comment (netcriptus) [inline requirements/requirements-dev.in:49]: Current version is now 1.30.1
Would it be worth it to check if it's good for us and update to the latest?

**Review by netcriptus — APPROVED** [review]
Just two minor comments, overall LGTM

File: raiden/routing.py [hunk 2]
```diff
@@ -61,18 +67,23 @@ def get_best_routes(
             )
 
             if is_usable is channel.ChannelUsability.USABLE:
-                # FIXME query address_metadata from direct endpoint on PFS
+                address_to_address_metadata = {Address(from_address): our_address_metadata}
```
[the remark this answers is not in the document]
Comment (ezdac) [reply raiden/routing.py:70]: You are correct, but I would leave it as is for consistency sake as of our developer guidelines

File: raiden/routing.py [hunk 3]
```diff
@@ -38,11 +39,16 @@ def get_best_routes(
     previous_address: Optional[Address],
     pfs_config: Optional[PFSConfig],
     privkey: PrivateKey,
+    our_address_metadata: AddressMetadata,
 ) -> Tuple[Optional[str], List[RouteState], Optional[UUID]]:
 
     token_network = views.get_token_network_by_address(chain_state, token_network_address)
     assert token_network, "The token network must be validated and exist."
 
+    if pfs_config is None or one_to_n_address is None:
```
Comment (fredo) [inline raiden/routing.py:48]: This ain't work just like this and is responsible for the errors in the CI. The tests failing include direct payments which did not need (and dont have) PFS config set. Now direct payments need to communicate to the PFS and thus need PFS config to be set.

**Review by fredo — CHANGES_REQUESTED** [review]

File: raiden/routing.py (hunk 3 above)
Comment (fredo) [reply raiden/routing.py:48]: Thinking about it, I come more and more to the conclusion that we should not require having a PFS config in place for direct payments. 
Same as we can accept routes with missing or incomplete metadata we should be able to send direct payments with wrong or missing metadata. 
Thus the above condition seems to me, should be at the place where it was before

Comment (fredo) [reply raiden/routing.py:48]: because we need a route for mediated payments which is essential for a mediated payment and in the case of direct payments we only need metadata which is not essential to a successful payment (we already know the route)

File: raiden/tests/integration/long_running/test_settlement.py [hunk 4]
```diff
@@ -505,6 +506,7 @@ def test_channel_withdraw(
         target=target,
         identifier=identifier,
         secret=secret,
+        route_states=[create_route_state_for_route([alice_app, bob_app], token_address)],
```
Comment (ezdac) [inline raiden/tests/integration/long_running/test_settlement.py:509]: why is this necessary? see general comment

**Review by ezdac — COMMENTED** [review]
Looks good!

Please help me understand, why is it necessary to pass the explicit `route_states` to the 
`mediated_transfer_async( )`? The way I see it this circumvents calls to the PFS being made and the whole `get_best_routes` codepath not being used. I don't see why the changes in the PR wouldn't handle querying of route-information.

File: raiden/routing.py [hunk 5]
```diff
@@ -69,7 +65,9 @@ def get_best_routes(
             if is_usable is channel.ChannelUsability.USABLE:
                 address_to_address_metadata = {Address(from_address): our_address_metadata}
                 try:
-                    address_metadata = query_address_metadata(pfs_config, to_address)
+                    address_metadata = None
+                    if pfs_config is not None:
+                        address_metadata = query_address_metadata(pfs_config, to_address)
                     if address_metadata:
```
[the remark this answers is not in the document]
Comment (fredo) [reply raiden/routing.py:71]: It is actually not necessary. That is the reason why I did not change it in every place in the tests. It will have two different outcomes. 

With route states: It has mocked address metadata and will not call `get_best_routes`
Without route states: It will call `get_best_routes` and will not query address metadata as `pfs_config is None` and leave the address_metadata field for its partner empty.
